build.sbt

```diff
@@ -25,13 +25,23 @@ lazy val xml = crossProject.in(file("."))
     scalacOptions         ++= "-deprecation:false -feature -Xlint:-stars-align,-nullary-unit,_".split("\\s+").to[Seq],
     scalacOptions in Test  += "-Xxml:coalescing",
 
-    apiMappings ++= Map(
-      scalaInstance.value.libraryJar
-        -> url(s"http://www.scala-lang.org/api/${scalaVersion.value}/"),
-      // http://stackoverflow.com/questions/16934488
-      file(System.getProperty("sun.boot.class.path").split(java.io.File.pathSeparator).filter(_.endsWith(java.io.File.separator + "rt.jar")).head)
-        -> url("http://docs.oracle.com/javase/8/docs/api")
-    )
+    apiMappings ++= {
+      Map(
+        scalaInstance.value.libraryJar
+          -> url(s"http://www.scala-lang.org/api/${scalaVersion.value}/")
+      ) ++ {
+        // http://stackoverflow.com/questions/16934488
+        Option(System.getProperty("sun.boot.class.path"))
+          .map(path =>
+            Map(
+              file(path
+                .split(java.io.File.pathSeparator)
+                .filter(_.endsWith(java.io.File.separator + "rt.jar")).head)
+              -> url("http://docs.oracle.com/javase/8/docs/api")
+            ))
+          .getOrElse(Map.empty)
+      }
+    }
```

Comment:
This refactoring actually causes an NPE to silently succeed with the `doc` task.  It's better if the task still fails, IMO.

This part of the sbt config may get replaced by https://github.com/ThoughtWorksInc/sbt-api-mappings

Reply:
Sorry, I take this back.  We will need to degrade when the Java boostrap classpath is unavailable.  I'll be using this PR as a basis for fixing Java 9 including the external doc linking.